feat(oxfmt): Introduce --write options

leaysgur opened the pull request:
Part of #10573

- Support `-w, --write` CLI flag
  - And also `--default` if not presented

Since I'm undecided whether to implement the exact same arguments as the Prettier CLI, so for now.

Boshen commented:
### Merge activity

* **Aug 29, 5:19 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
feat(oxfmt): Introduce --write options (#13383)

Part of #10573

- Support `-w, --write` CLI flag
  - And also `--default` if not presented

Since I'm undecided whether to implement the exact same arguments as the Prettier CLI, so for now.
